javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java

```diff
@@ -126,6 +126,7 @@ public ProxyMethod map(Operation operation) {
         List<ProxyMethodParameter> parameters = new ArrayList<>();
         for (Parameter parameter : operation.getRequest().getParameters().stream()
                 .filter(p -> p.getProtocol() != null && p.getProtocol().getHttp() != null).collect(Collectors.toList())) {
+            parameter.setOperation(operation);
```

Comment:
This should be handled in the preprocessor.

Reply:
I read your comments. Scratch my comment above. I'm okay with this change for now but this does expose a defection in the design.

Reply:
Yes, it is more or less design issue.

Either it can be copied when changed (I've got a shallowCopy util in fluentgen to do this https://github.com/weidongxu-microsoft/autorest.java/blob/v4_fluentgen/fluentgen/src/main/java/com/azure/autorest/fluent/util/Utils.java#L34), or need other refactor elsewhere.

Currently I prefer not to do bigger change (this will just invite large merge conflict) until I've merged javagen and optionally fluentgen and your pending big refactor.